[RFR] Backward compatibility for merge with original repository

djhi opened the pull request:
As requested on https://github.com/tj/node-ratelimiter/pull/21, I made the following changes:
- [x] Use 2 spaces instead of 4
- [x] Added backward compatibility where possible (`Limiter` initialization) with deprecation warning
- [x] Added back old tests to ensure backward compatibility (except for old `get` which is now `newHit`)
- [x] Updated README with instructions for migration
- [x] Removed `stage-0` which was unnecessary
- [x] Fixed a bug in `Limiter` resulting in double increase of hits

I also removed compiled files (in `lib` from git as those will only be used for npm publication.

djhi committed:
Use 2 spaces instead of 4
added lib to gitignore

djhi committed:
Added upgrade instructions. Updated package.json

djhi committed:
Added backward compatibility with warnings and further updated README with migration instructions

Kmaschta commented:
Need rebase please

Kmaschta commented on package.json:
```diff
@@ -27,7 +27,8 @@
   "contributors": [
-    "Francois-Guillaume Ribreau <npm@fgribreau.com>"
+    "Francois-Guillaume Ribreau <npm@fgribreau.com>",
+    "Gildas Garcia (@marmelab) <gildas@marmelab.com"
```
s/<gildas@marmelab.com"/<gildas@marmelab.com>"/

Kmaschta replied:
s/.com"/.com>"/

Kmaschta commented on src/index.spec.js:
```diff
@@ -0,0 +1,332 @@
+          duration: 10000,
+          max: 2,
+          id: 'something',
+          db: db
```
s/db: db/db,/

marmelab-bot committed:
Typo fix s/db: db/db,/

As requested by @Kmaschta at https://github.com/marmelab/node-ratelimiter/pull/5#discussion_r76823188

Kmaschta commented:
:+1: 

Not sure about this commit 6f83fdb57f07cf175594b6637ec4337e41dc1a3b, is the tests ES6-compatible?

Also, we can add @jpetitcolas to the contributors since he fixed some bugs.
